### PR TITLE
[BUGFIX] _npi_repeats with swap

### PR DESCRIPTION
**Pull request by Kacper-Pietkun**
## Description ##
_npi_repeats gave correct outputs only when repetition happened on 0th axis. That's why swapaxes function was called before and after triggering _npi_repeats operator. swapaxes() was called from python side, thus when _npi_repeats was a node in a computational graph it resulted in wrong output (swapaxes wasn't called in that case).

## Solution ##
Trigger SwapAxis operator directrly by _npi_repeats operator.

### Essentials ###
- [x] PR's title starts with a category (e.g. [BUGFIX], [MODEL], [TUTORIAL], [FEATURE], [DOC], etc)
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage
- [x] Code is well-documented

## Comments ##
- Added check for repeats parameter - it throws ValueError when length of the list is not equal to the number of elements on the axis that is going to be repeated (numpy compatibility)

### CODE REVIEW

File: src/operator/numpy/np_repeat_op-inl.h [hunk 1]
```diff
@@ -219,15 +225,86 @@ void NumpyRepeatsOpForward(const nnvm::NodeAttrs& attrs,
...
+  if (axisOpt.has_value() && axisOpt.value() != 0) {
+    int type_size          = mshadow_sizeof(inputs[0].type_flag_);
+    size_t total_temp_size = inputs[0].shape_.Size() * type_size +
+                             outputs[0].shape_.Size() * type_size + repts.ndim() * sizeof(int);
+    Tensor<xpu, 1, char> temp_space = ctx.requested[0].get_space_typed<xpu, 1, char>(
+        Shape1(total_temp_size), ctx.get_stream<xpu>());
+    void* swap_output_tmp_dptr   = temp_space.dptr_;
+    void* repeat_output_tmp_dptr = temp_space.dptr_ + inputs[0].shape_.Size() * type_size;
+    int* repeat_tmp_dptr =
+        reinterpret_cast<int*>(temp_space.dptr_ + inputs[0].shape_.Size() * type_size +
```
Comment (bgawrych) [inline src/operator/numpy/np_repeat_op-inl.h:271]:
```suggestion
        reinterpret_cast<int*>(repeat_output_tmp_dptr +
```

**Review by bgawrych — APPROVED** [review]

**Comment by Kacper-Pietkun** [comment]
@mxnet-bot run ci [centos-gpu]

**Comment by Kacper-Pietkun** [comment]
@mxnet-bot run ci [unix-gpu, website]

**Comment by Kacper-Pietkun** [comment]
@mxnet-bot run ci [centos-gpu]

**Comment by Kacper-Pietkun** [comment]
@mxnet-bot run ci [centos-gpu, unix-gpu, website]

**Comment by Kacper-Pietkun** [comment]
@mxnet-bot run ci [centos-gpu, unix-gpu, website]

**Comment by Kacper-Pietkun** [comment]
@mxnet-bot run ci [centos-gpu]

File: src/operator/numpy/np_repeat_op-inl.h [hunk 2]
```diff
@@ -102,6 +103,14 @@ inline bool RepeatsOpShape(const nnvm::NodeAttrs& attrs,
     return true;
   }
 
+  mxnet::Tuple<int> repts = param.repeats.value();
```
Comment (bartekkuncer) [inline src/operator/numpy/np_repeat_op-inl.h:106]: Can you use more descriptive variable names? Without abbreviations if possible.

File: src/operator/numpy/np_repeat_op-inl.h [hunk 3]
```diff
@@ -153,26 +162,27 @@ struct repeat_axis_fwd {
 };
 
 template <typename xpu>
-void NumpyRepeatsOpForward(const nnvm::NodeAttrs& attrs,
-                           const OpContext& ctx,
-                           const std::vector<TBlob>& inputs,
-                           const std::vector<OpReqType>& req,
-                           const std::vector<TBlob>& outputs) {
+void NumpyRepeatsAxisZeroOpForward(const nnvm::NodeAttrs& attrs,
```
Comment (bartekkuncer) [inline src/operator/numpy/np_repeat_op-inl.h:165]: Please add brief description of this function.

File: src/operator/numpy/np_repeat_op-inl.h [hunk 4]
```diff
@@ -153,26 +162,27 @@ struct repeat_axis_fwd {
...
-    return;
-  Stream<xpu>* s = ctx.get_stream<xpu>();
-
-  int repeats = 0;
+  int repeats                 = 0;
   dmlc::optional<int> axisOpt;
-  int axis                  = -1;
+  // axis is always set to zero, because before calling this function target axis of the repeat
+  // function is swapped with 0th axis
+  int axis                  = 0;
```
Comment (bartekkuncer) [inline src/operator/numpy/np_repeat_op-inl.h:178]: constexpr

File: src/operator/swapaxis-inl.h [hunk 5]
```diff
@@ -156,7 +156,7 @@ void SwapAxisCompute(const nnvm::NodeAttrs& attrs,
                      const std::vector<OpReqType>& req,
                      const std::vector<TBlob>& out_data) {
   using namespace mshadow;
-  MSHADOW_TYPE_SWITCH(in_data[swapaxisenum::kData].type_flag_, DType, {
+  MSHADOW_TYPE_SWITCH_EXT_WITH_BOOL(in_data[swapaxisenum::kData].type_flag_, DType, {
```
Comment (bartekkuncer) [inline src/operator/swapaxis-inl.h:159]: Maybe it is good time to change swapaxisenum to just swapaxis as it is for other ops?

**Review by bartekkuncer — CHANGES_REQUESTED** [review]

**Comment by Kacper-Pietkun** [comment]
@mxnet-bot run ci [website]

**Review by bartekkuncer — APPROVED** [review]
LGTM

**Review by bartekkuncer — APPROVED** [review]

**Pull request merged**